### PR TITLE
Clean-ups

### PR DESCRIPTION
**Pull request by piranna**
Multiple clean-ups I've found while working with the mediasoup source code. Most of them are simple and automated ones, so they are safely to merge. The list of changes is:

- [Update deprecated Github Actions](https://github.com/versatica/mediasoup/commit/28b90c0943d16a989e1f1b4d475831168382c917)
- [Linted markdown and typescript files](https://github.com/versatica/mediasoup/commit/b267f8f9afaa950db4ada316f7eb01d74e6ee90c)
- [Sorted package.json files and scripts](https://github.com/versatica/mediasoup/commit/b99b4665adec051d1e097cc7e49d70c45ac6f665)
- [Added packaged modules to .gitignore file](https://github.com/versatica/mediasoup/commit/2d3e9f8fe2e9a50a6536fa3cbeca217b5489d6fe)
- [Reuse BUILD_DIR variable in makefile](https://github.com/versatica/mediasoup/commit/672631e3693ea7aa83c11f335bdb96c4a338bccb)

### CODE REVIEW

File: .gitignore
Comment (ibc) [inline .gitignore]:
```suggestion
# Packaged module.
```

What is a packaged module?

File: .github/workflows/mediasoup-rust.yaml
Comment (ibc) [inline .github/workflows/mediasoup-rust.yaml:22]: 👍

File: doc/Building.md
Comment (ibc) [inline doc/Building.md:5]: Why all these changes? There are two blank lines on purpose.

File: doc/Building.md
Comment (ibc) [inline doc/Building.md:104]: Same here. Why are you removing the `$` symbol? It's cosmetic and super opinionated, but we put it on purpose.

File: node/src/Worker.ts
Comment (ibc) [inline node/src/Worker.ts:186]: I fail to understand why my editor is not removing those spaces automatically on save. It usually does. Good catch.

File: package.json
Comment (ibc) [inline package.json]: Is it just about sorting items alphabetically? If so, @nazar-pc: is this what you asked for?

File: package.json
Comment (ibc) [inline package.json]: Definitely I don't like these changes. Order of scripts was set on purpose. I don't want that those are ordered alphabetically. Let's revert this, please.

File: .gitignore
Comment (piranna) [reply .gitignore]: When you exec [npm pack](https://docs.npmjs.com/cli/v7/commands/npm-pack) manually and not as part of `npm publish`, it generates a tarfile, that `*.tgz` file that I've put in the `.gitignore` file. I usually do that to check what's actually being included in the package itself that's going to be published, since it's a good practice to don't add test files or similar things. The tarfile can be also served from a vanilla HTTP server or installed standalone, for example, but you don't want to upload it to the git repo since it's already a copy of the repo code.

File: doc/Building.md
Comment (piranna) [reply doc/Building.md:5]: Not fully sure, when saving the file in VS Code, `markdownlint` default rules automatically remove them, the same as with the `$` on the shell code snippets. I can revert that changes if you prefer it.

File: .gitignore
Comment (ibc) [reply .gitignore]: ok

File: node/src/Worker.ts
Comment (piranna) [reply node/src/Worker.ts:186]: Give thanks to VS Code, not me :-)

Comment (piranna) [reply node/src/Worker.ts:186]: Maybe should we run a full check / clean-up for these typos in all the code?

File: doc/Building.md
Comment (ibc) [reply doc/Building.md:5]: I'm fine with the removal of the two consecutive blank lines, but I do not want that those '$'  are removed. Why would CS Code remove them on file save?

File: package.json
Comment (piranna) [reply package.json]: Yes, only sort them, so it's easier to identify later both additions and deletions.

File: node/src/Worker.ts
Comment (ibc) [reply node/src/Worker.ts:186]: I expected our ESLint tules to catch them but they don't.

File: package.json
Comment (piranna) [reply package.json]: It was the same, to easily identify changes, but ok, I'll revert it.

File: doc/Building.md
Comment (piranna) [reply doc/Building.md:5]: No idea, I was a bit surprised too, I think the lint rule is about the `$` is superfluos and don't allow just copy&paste the code block content.

File: node/src/Worker.ts
Comment (piranna) [reply node/src/Worker.ts:186]: Yeah, that surprised me a bit too.

File: doc/Building.md
Comment (ibc) [reply doc/Building.md:5]: Which rule? 😀 We don't enforce any rule in `.md` files. If your VS Code does it... well ¯\_(ツ)_/¯

File: package.json
Comment (ibc) [reply package.json]: Thanks.

File: doc/Building.md
Comment (piranna) [reply doc/Building.md:5]: Yes, VS Code has its own opinions too... ;-) I've tried to find the markdownlint rule but can't find it, maybe I looked the wrong one, but I swear I've seen it was a bad practice... but yes, I've not done the change myself, so no issues, just only now I don't know why it happened 😅

File: package.json
Comment (piranna) [reply package.json]: Done.

File: package.json
Comment (nazar-pc) [reply package.json]: Directories should go first (all of them), then files (starting with those that are capitalized if any). Basically should be the same order you typically see it in the repository or file manager. Just open https://github.com/versatica/mediasoup and see how files are listed.

**Review by nazar-pc — COMMENTED** [review]
I'm fine with deleting `$` BTW, we don't have `#` to distinguish from root commands anyway, so it is kind of redundant.

File: doc/Building.md
Comment (piranna) [reply doc/Building.md:104]: Found the rule: https://github.com/DavidAnson/markdownlint/blob/v0.27.0/doc/md014.md

> Rationale: It is easier to copy/paste and less noisy if the dollar signs are omitted when they are not needed. See https://cirosantilli.com/markdown-style-guide#dollar-signs-in-shell-code for more information.

If you still wants them, I can revert them back, and maybe add a `.markdownlintrc` file to prevent happening again, the same for the two empty lines if you want.

**Comment by piranna** [comment]
> I'm fine with deleting $ BTW, we don't have # to distinguish from root commands anyway, so it is kind of redundant.

I can do whatever both of you decide, remove them, add them back, or add them back and add a `.markdownlintrc` file. I prefer remove them too.


> Directories should go first (all of them), then files (starting with those that are capitalized if any). Basically should be the same order you typically see it in the repository or file manager. Just open [versatica/mediasoup](https://github.com/versatica/mediasoup) and see how files are listed.

So, alphabetic order, but with directories first, isn't it? Ok, I can do that, at least provides some order.

**Comment by piranna** [comment]
```json
    "node/lib",
    "worker/deps/libwebrtc",
    "worker/fuzzer/include",
    "worker/fuzzer/src",
    "worker/include",
    "worker/src",
    "worker/test/include",
    "worker/test/src",
    "npm-scripts.js",
    "worker/Makefile",
    "worker/meson.build",
    "worker/meson_options.txt",
    "worker/scripts/*.js",
    "worker/scripts/*.json",
    "worker/scripts/*.py",
    "worker/scripts/*.sh",
    "worker/subprojects/*.wrap"
```

Is this order fine?

**Comment by nazar-pc** [comment]
Just confirm with GitHub, no need to ask me.

Also please leave multiple comments as review rather that separately, I've got 28 email notifications about this PR already! This is not sustainable, a bit more discipline goes a long way to save maintainer's sanity.

**Comment by ibc** [comment]
I will rereview and answer questions tomorrow. Thanks.

**Comment by ibc** [comment]
> ```json
>     "node/lib",
>     "worker/deps/libwebrtc",
>     "worker/fuzzer/include",
>     "worker/fuzzer/src",
>     "worker/include",
>     "worker/src",
>     "worker/test/include",
>     "worker/test/src",
>     "npm-scripts.js",

"npm-scripts.js" cannot be between "worker/test/src" and "worker/Makefile". I'd move it to the end.

>     "worker/Makefile",
>     "worker/meson.build",
>     "worker/meson_options.txt",
>     "worker/scripts/*.js",
>     "worker/scripts/*.json",
>     "worker/scripts/*.py",
>     "worker/scripts/*.sh",
>     "worker/subprojects/*.wrap"
> ```
> 
> Is this order fine?

**Comment by ibc** [comment]
> "npm-scripts.js" cannot be between "worker/test/src" and "worker/Makefile". I'd move it to the end.

Actually I'd follow this: https://github.com/versatica/mediasoup/pull/1003#discussion_r1094925453

**Comment by ibc** [comment]
> I'm fine with deleting `$` BTW, we don't have `#` to distinguish from root commands anyway, so it is kind of redundant.

`$` for me means "shell or bash command" (without entering into details about whether it's a normal user or root user). But ok, let's remove it.

**Comment by piranna** [comment]
> "npm-scripts.js" cannot be between "worker/test/src" and "worker/Makefile". I'd move it to the end.

Having all folders at begin, and later all files, and each group in alphabetical order, it gets like that. BUT, grouping folders and files inside each sub-folder, and putting (sub)folders first, it gets like this:

```js
    "node/lib",
    "worker/deps/libwebrtc",
    "worker/fuzzer/include",
    "worker/fuzzer/src",
    "worker/include",
    "worker/scripts/*.js",
    "worker/scripts/*.json",
    "worker/scripts/*.py",
    "worker/scripts/*.sh",
    "worker/src",
    "worker/subprojects/*.wrap",
    "worker/test/include",
    "worker/test/src",
    "worker/Makefile",
    "worker/meson.build",
    "worker/meson_options.txt",
    "npm-scripts.js"
```

It makes total sense to me, it's more alike to how VS Code file explorer does, so whatever you want.

File: package.json
Comment (piranna) [reply package.json]: Like this?

```js
    "node/lib",
    "worker/deps/libwebrtc",
    "worker/fuzzer/include",
    "worker/fuzzer/src",
    "worker/include",
    "worker/scripts/*.js",
    "worker/scripts/*.json",
    "worker/scripts/*.py",
    "worker/scripts/*.sh",
    "worker/src",
    "worker/subprojects/*.wrap",
    "worker/test/include",
    "worker/test/src",
    "worker/Makefile",
    "worker/meson.build",
    "worker/meson_options.txt",
    "npm-scripts.js"
```

**Comment by ibc** [comment]
Works for me. I'll come back to review this on next week (too busy until then).

**Comment by piranna** [comment]
> Works for me. I'll come back to review this on next week (too busy until then).

Done.

**Review by ibc — APPROVED** [review]
Got some time. LGTM

**Comment by ibc** [comment]
Waiting for CI

**Review by jmillan — APPROVED** [review]

**Pull request merged**
